Adding adding optional parameters to search and retrieve functions

chinacat567 opened the pull request:
### Change
- Passing an optional parameters to search and retrieve functions in a dict
- These parameters are then being passed to the HTTP requests in ``_http_get_application_json `` and `` _http_get_multipart_application_dicom `` 

### Testing
- ``pytest --flake8``

chinacat567 committed:
adding optional params to search and retrieve functions

medihack reviewed (commented):
- Is there a reason for the different naming in the search methods ("additional_params") vs the retrieve methods (just "params")? How about naming it in the same way (I think "additional_params" is fine).
- What about the other methods like "retrieve_instance_frames", "retrieve_instance_metadata", ... Wouldn't it make sense to add the additional params there, too?
- Also some simple tests would be nice (added to test_web.py) that checks that an additional param is passed through.

chinacat567 committed:
Added tests and additional params to more methods

chinacat567 committed:
adding params to retrieve_instance_frames

chinacat567 commented:
### Revision
- Renamed the argname to ``additional_params`` in all methods for consistency 
- Added ``additional_params`` to more get methods
- Updated the tests

chinacat567 committed:
fixing broken tests
